pymc_marketing/mmm/fourier.py

```diff
@@ -422,25 +424,62 @@ def sample_prior(self, coords: dict | None = None, **kwargs) -> xr.Dataset:
         coords[self.prefix] = self.nodes
         return self.prior.sample_prior(coords=coords, name=self.variable_name, **kwargs)
 
-    def sample_curve(self, parameters: az.InferenceData | xr.Dataset) -> xr.DataArray:
-        """Create full period of the fourier seasonality.
+    def sample_curve(
+        self,
+        parameters: az.InferenceData | xr.Dataset,
+        use_dates: bool = False,
+        start_date: datetime.datetime | None = None,
+    ) -> xr.DataArray:
+        """Create full period of the Fourier seasonality.
 
         Parameters
         ----------
         parameters : az.InferenceData | xr.Dataset
-            Inference data or dataset containing the fourier parameters.
+            Inference data or dataset containing the Fourier parameters.
             Can be posterior or prior.
+        use_dates : bool, optional
+            If True, use datetime coordinates for the x-axis. Defaults to False.
+        start_date : datetime.datetime, optional
+            Starting date for the Fourier curve. If not provided and use_dates is True,
+            it will be derived from the current year or month. Defaults to None.
 
         Returns
         -------
         xr.DataArray
-            Full period of the fourier seasonality.
+            Full period of the Fourier seasonality.
 
         """
-        full_period = np.arange(self.days_in_period + 1)
-        coords = {
-            "day": full_period,
-        }
+        # Determine the full period
+        full_period = np.arange(int(self.days_in_period + 1.5))
+
+        coords = {}
+        if use_dates:
+            if start_date is None:
+                # Derive start_date based on the type of Fourier seasonality
+                today = datetime.datetime.now()
+                if isinstance(self, YearlyFourier):
+                    start_date = datetime.datetime(year=today.year, month=1, day=1)
+                elif isinstance(self, MonthlyFourier):
+                    start_date = datetime.datetime(
+                        year=today.year, month=today.month, day=1
+                    )
```

Comment:
Is it possible to not rely on the child classes? This makes an inheritance from the `FourierBase` not possible.

This can usually be solved by a method in each child class. For instance, `get_default_start_date(today: datetime)`